Sources/Sentry/SentryTracer.m

```diff
@@ -92,6 +103,14 @@ - (instancetype)initWithTransactionContext:(SentryTransactionContext *)transacti
             initSlowFrames = currentFrames.slow;
             initFrozenFrames = currentFrames.frozen;
         }
+#endif
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        [profilerLock lock];
```

Comment:
`m`: Can we add a test for this new logic or did I miss that?

Reply:
`testCapturesProfile_whenProfilingEnabled` and `testDoesNotCapturesProfile_whenProfilingDisabled` in `SentryTracerTests` should cover this.